dpnp/dpnp_algo/dpnp_algo_mathematical.pyx

```diff
@@ -139,54 +141,28 @@ cpdef dparray dpnp_cross(dparray x1, dparray x2):
     return call_fptr_2in_1out(DPNP_FN_CROSS, x1, x2, x1.shape)
 
 
-cpdef dparray dpnp_cumprod(dparray x1, bint usenan=False):
+cpdef dparray dpnp_cumprod(dparray x1):
+    #instead of x1.shape, (x1.size, ) is passed to the function
+    # due to the following:
+    # >>> import numpy
+    # >>> a = numpy.array([[1, 2], [2, 3]])
+    # >>> res = numpy.cumprod(a)
+    # >>> res.shape
+    # (4,)
 
-    types_map = {
-        dpnp.int32: dpnp.int64,
-        dpnp.int64: dpnp.int64,
-        dpnp.float32: dpnp.float32,
-        dpnp.float64: dpnp.float64
-    }
-
-    res_type = types_map[x1.dtype.type]
-
-    cdef dparray result = dparray(x1.size, dtype=res_type)
-
-    cur_res = 1
-
-    for i in range(result.size):
-
-        if not usenan or not dpnp.isnan(x1[i]):
-            cur_res *= x1[i]
-
-        result._setitem_scalar(i, cur_res)
-
-    return result
-
-
-cpdef dparray dpnp_cumsum(dparray x1, bint usenan=False):
-
-    types_map = {
-        dpnp.int32: dpnp.int64,
-        dpnp.int64: dpnp.int64,
-        dpnp.float32: dpnp.float32,
-        dpnp.float64: dpnp.float64
-    }
-
-    res_type = types_map[x1.dtype.type]
-
-    cdef dparray result = dparray(x1.size, dtype=res_type)
-
-    cur_res = 0
-
-    for i in range(result.size):
+    return call_fptr_1in_1out(DPNP_FN_CUMPROD, x1, (x1.size,))
 
-        if not usenan or not dpnp.isnan(x1[i]):
-            cur_res += x1[i]
 
-        result._setitem_scalar(i, cur_res)
+cpdef dparray dpnp_cumsum(dparray x1):
+    #instead of x1.shape, (x1.size, ) is passed to the function
```

Comment:
Please, make this comment section as a docstring, if it is needed or just remove it.